src/System.Private.ServiceModel/src/System/ServiceModel/Channels/ServiceModelHttpMessageHandler.cs

```diff
@@ -27,5 +29,13 @@ public CookieContainer CookieContainer
             get { return _innerHandler.CookieContainer; }
             set { _innerHandler.CookieContainer = value; }
         }
+
+        protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            var content = request.Content as MessageContent;
+            var responseMessage = await base.SendAsync(request, cancellationToken);
+            await content.WriteCompletionTask;
```

Comment:
Normally libraries use ```ConfigureAwait(false)``` with awaits.  Does WCF not need to worry about that for some reason, e.g. it'll never be used with non-null SynchronizationContext, custom TaskSchedulers, etc.?